album_folder.py

```diff
@@ -4,27 +4,115 @@
 import progressbar
 import os
 import sys
+import time
+from argparse import ArgumentParser
 from shutil import copyfile
+from shutil import move
+
+
+# Source_dir : passed as parameter, where your photos are located
+# output_dir : directory under where all albuns will be created
 
 def run(source_dir, output_dir):
     bar = progressbar.ProgressBar()
-    for f in bar(os.listdir(source_dir)):
-        if os.path.splitext(f)[1] == '.json':
+
+    if not os.path.exists(output_dir):
+        os.mkdir(output_dir)
+
+    # Let's print some statistics
+    num_of_missing = 0
+    num_of_albuns = 0
+    num_of_copied = 0
+    num_of_moved = 0
+    num_of_total_files = 0
+    num_of_json = 0
+
+    all_files = os.listdir(source_dir)
+    num_of_total_files = len(all_files)
+
+    for f in bar(all_files):                                       # f is each file in directory
+        (root_file, ext) = os.path.splitext(f)
+        if ext == '.json':                                                      # only .json are treated
```

Comment:
Instead of having the entire body of the loop in a giant if statement, instead do:
```python
if ext != '.json':
    continue
```
Or, make the body of the loop a function. Either way, the maximum nesting level is reduced and the code is easier to read.

Also, the comment "only .json are treated" doesn't add any value; the if statement clearly states the same thing.

Reply:
OK. I'm changing. 
But just understant, I'm from oldies of programming, and it has a long, long time I do not program. I agree that it simplifies the code reading, but my formation was that statements like continue, pass, and others that breaks the flow of the execution are to be avoided. 
But I'm new on python, I learned it by myself, just for fun, and those details, I still miss some....
att,...